pilot/pkg/model/push_context.go

```diff
@@ -695,6 +695,21 @@ func (ps *PushContext) ServiceForHostname(proxy *Proxy, hostname host.Name) *Ser
 	return nil
 }
 
+// ServicesForHostname returns a list of services that fall under the hostname provided. This hostname
+// can be a wildcard.
+func (ps *PushContext) ServicesForHostname(proxy *Proxy, hostname host.Name) []*Service {
```

Comment:
See the changes I am making in https://github.com/istio/istio/pull/34694. Pass SidecarScope and sidecar to get this not proxy. We will remove dependency of proxy in cluster/cluster builder.

Reply:
no longer applicable